examples/shader.rs

```diff
@@ -0,0 +1,194 @@
+            .with_gl(GlRequest::Specific(Api::OpenGlEs, (3, 0)))
+            .with_vsync(false)
+            .build_windowed(wb, &el)
+            .unwrap();
+        let windowed_context = unsafe { windowed_context.make_current().unwrap() };
+        windowed_context
+    };
+
+    let context: Context;
+    let framebuffer: NativeFramebuffer;
+    let texture_colorbuffer: NativeTexture;
+    let shader_program: Program;
+    let mut renderer: OpenGl;
+
+    unsafe {
+        context = glow::Context::from_loader_function(|symbol| windowed_context.get_proc_address(symbol) as *const _);
+        renderer = OpenGl::new_from_function(|symbol| windowed_context.get_proc_address(symbol) as *const _).unwrap();
+
+        shader_program = create_shader_program(&context);
+        (framebuffer, texture_colorbuffer) = create_framebuffer_colorbuffer(&context);
+
+        renderer.set_screen_target(Some(framebuffer));
+    }
+
+    let mut canvas = Canvas::new(renderer).expect("Cannot create canvas");
+    canvas.set_size(WINDOW_WIDTH as u32, WINDOW_HEIGHT as u32, 1.0);
+
+    el.run(move |event, _, control_flow| {
+        *control_flow = ControlFlow::Poll;
+
+        match event {
+            Event::LoopDestroyed => return,
+            Event::WindowEvent { ref event, .. } => match event {
+                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
+                _ => {}
+            },
+            Event::RedrawRequested(_) => {
+                // draw red rectangle on white background
+
+                let dpi_factor = windowed_context.window().scale_factor();
+                let size = windowed_context.window().inner_size();
+                canvas.set_size(size.width as u32, size.height as u32, dpi_factor as f32);
+                canvas.clear_rect(0, 0, size.width as u32, size.height as u32, Color::rgbf(1., 1., 1.));
+
+                prepare_framebuffer_for_render(&context, framebuffer);
+
+                canvas.save();
+
+                let paint = Paint::color(Color::rgbf(1., 0., 0.));
+                let mut path = Path::new();
+                path.rect(WINDOW_WIDTH / 2. - 25., WINDOW_HEIGHT / 2. - 25., 50., 50.);
+                canvas.fill_path(&mut path, paint);
+                canvas.restore();
+
+                canvas.flush();
+
+                // shader inverts colors
+                render_framebuffer_to_screen(&context, shader_program, texture_colorbuffer);
+
+                windowed_context.swap_buffers().unwrap();
+            }
+            Event::MainEventsCleared => windowed_context.window().request_redraw(),
+            _ => (),
+        }
+    });
+}
+
+fn create_shader_program(context: &glow::Context) -> NativeProgram {
+    unsafe {
+        let v_shader = context.create_shader(glow::VERTEX_SHADER).unwrap();
+        let vert_shader: String = fs::read_to_string("assets/screen.vert.glsl").unwrap().parse().unwrap();
+        context.shader_source(v_shader, vert_shader.as_str());
```

Comment:
Can you try and see if this trick works? I think that it might be more idiomatic to include the shaders at compile time. The same could be done for the fragment shader as well then.

```suggestion
        let vert_shader = include_str!("assets/screen.vert.glsl");
        context.shader_source(v_shader, vert_shader);
```

Reply:
Yeah, that works! Nice!